src/NServiceBus.Core/Serializers/MessageDeserializerResolver.cs

```diff
@@ -12,14 +12,17 @@ class MessageDeserializerResolver
 
         public MessageDeserializerResolver(IEnumerable<IMessageSerializer> messageSerializers, Type defaultSerializerType)
         {
-            Guard.AgainstNull("defaultSerializerType", defaultSerializerType);
-
             serializersMap = messageSerializers.ToDictionary(key => key.ContentType, value => value);
             defaultSerializer = serializersMap.Values.Single(serializer => serializer.GetType() == defaultSerializerType);
         }
 
         public IMessageSerializer Resolve(string contentType)
         {
+            if (contentType == null)
```

Comment:
I would instead validate this method parameter against null and expose a more explicit api, such as `DefaultDeserializer` property on this class.
Then is up to the consumer to call the `DefaultDeserializer` when content type is `null`.